### DISCUSSION TITLE
文本归一化模块添加参数选择保留感叹词或者去掉感叹词

### DISCUSSION THREAD

**Discussion by Shengqiang-Li** [fishaudio/Bert-VITS2#309] [Ideas]
在目前的pipeline中，使用WetextProcessing做文本归一化时，默认会去掉感叹词，从而导致推理时感叹词不能发音，例如“啊”。有一个解决办法是使用WetextProcessing的输入参数来控制是否要去掉感叹词。我已经解决了这个问题，并计划提个PR。

**Comment by Stardust-minus** [comment]
可以的

**Comment by ruby11dog** [comment]
> 在目前的pipeline中，使用WetextProcessing做文本归一化时，默认会去掉感叹词，从而导致推理时感叹词不能发音，例如“啊”。有一个解决办法是使用WetextProcessing的输入参数来控制是否要去掉感叹词。我已经解决了这个问题，并计划提个PR。

请问解决了吗？wetextProcessing正则化会去掉感叹词的问题，我也遇到了

**Comment by Stardust-minus** [comment]
> > 在目前的pipeline中，使用WetextProcessing做文本归一化时，默认会去掉感叹词，从而导致推理时感叹词不能发音，例如“啊”。有一个解决办法是使用WetextProcessing的输入参数来控制是否要去掉感叹词。我已经解决了这个问题，并计划提个PR。
> 
> 请问解决了吗？wetextProcessing正则化会去掉感叹词的问题，我也遇到了

目前已经更新，现在文本归一化部分会保留感叹词了

**Reply by ruby11dog** [reply]
是怎么解决的呢？能给一下修改的代码地址嘛？thx

**Reply by Stardust-minus** [reply]
https://github.com/fishaudio/Bert-VITS2/commit/3cebcde0e35b061b80811d8bda291ec294473994

**Reply by yubinyes** [reply]
你好，我这两天刚训练的，按照webui_preprocess.py的步骤，我看text/chinese.py第十二行，已经改成normalizer = Normalizer(remove_interjections=False, remove_erhua=False).normalize了。我用自己训练的模型读取中文文本，最后一个字是“啊”的话，仍然会跳过去，请问还有其他地方需要修改吗？ （但是像阿，锕的话可以读出来）
